pkg/backend/backend.go

```diff
@@ -7,14 +7,24 @@ import (
 
 	"github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/locksutil"
 	"github.com/hashicorp/vault/sdk/logical"
 	"github.com/puppetlabs/vault-plugin-secrets-oauthapp/pkg/provider"
 )
 
 type backend struct {
+	ctx              context.Context
```

Comment:
We typically should avoid putting a context on a struct because it can lead to some weird behavior if others aren't aware of it. I see we also pass in context to some of the backend methods, but create a cancelation derived from the struct context, is there any reason we don't just wrap the backend struct in a control loop that takes context from the entrypoint? I'm guessing it's some limitation of vault's plugin api?

Reply:
Know what's funny? https://github.com/puppetlabs/relay-api/pull/495#discussion_r543686052

Reply:
I appreciate the irony of that, yeah. :-)

So the difficulty here is indeed with the plugin API that Vault provides -- basically I need that outer, global context when the config path is written to, and the context passed in to that method is rather request-scoped. However I think I have a way that I can just propagate the cancellation function instead of the context itself, which might make the code a bit less brittle.